doc/api.rst

```diff
@@ -134,6 +137,87 @@ call to ``grid_init`` for instance) before it is passed into this
 constructor.
 
 
+Device infrastructure attributes
+################################
+
+The fields have some infrastructure capabilities to allow the allocation of
+the data in different memory regions (usually acceleration devices but it
+can also be a user provided data layout on the same host) and manage the
+synchronization between the original data and the device data.
+
+These capabilities are provided by the following field attributes:
+
+ - `field_type%data_on_device`: A boolean to indicate if the data has already
+   been allocated and copied in the device.
+
+ - `field_type%read_from_device_f` or `field_type%read_from_device_c`: Function
+   pointers that provide the synchronization method to copy the data back from
+   the device into the host. The user needs to provide on one of these function
```

Comment:
Should this say "...provide only one of..."? If so, I'll fix it and commit...

Reply:
Ups, I wanted to say "...provide one of ..." but "only one" also works.